### PR TITLE
General updates

### PR DESCRIPTION
**Pull request by LiamGoodacre**
Removed docs and doc generation in favour of pursuit link in README.
No longer need rimraf - we were using it to remove docs.
Updated version of pulp.
Fixed most of the warnings (class imports, sections, operator aliases, etc).

NOTE: I had to come up with real names for operators! - why can't an alias be exported without exporting the real name?  (See Setter.purs for example)

### CODE REVIEW

**Comment by garyb** [comment]
> why can't an alias be exported without exporting the real name?

Aliases are replaced with the real names during desugaring, so the real name needs the same accessibility level as the alias.

File: src/Data/Lens/Setter.purs [hunk 1]
```diff
@@ -56,36 +72,29 @@ iover l f = l (Indexed $ uncurry f)
...
 
-(++~) :: forall s t a. (Semigroup a) => Setter s t a a -> a -> s -> t
-(++~) p = over p <<< flip append
+appendOver :: forall s t a. (Semigroup a) => Setter s t a a -> a -> s -> t
+appendOver p = over p <<< flip append
 
-(?~) :: forall s t a b. Setter s t a (Maybe b) -> b -> s -> t
-(?~) p = set p <<< Just
+maybeSet :: forall s t a b. Setter s t a (Maybe b) -> b -> s -> t
+maybeSet p = set p <<< Just
```
Comment (garyb) [inline src/Data/Lens/Setter.purs]: `setJust` perhaps?

**Comment by LiamGoodacre** [comment]
Ah messed up - hold on

**Comment by garyb** [comment]
I'd suggest using something this for the `package.json`, by the way: https://github.com/purescript/purescript-identity/blob/master/package.json - so rimraf is still somewhat useful :)

**Comment by garyb** [comment]
Likewise the `.travis.yml` from `purescript-identity` - it uses a container image with more juice for much faster builds.

**Comment by LiamGoodacre** [comment]
Is the `.travis.yml` change just a case of copy paste from there then?

**Comment by garyb** [comment]
Yeah, it should be.

**Comment by LiamGoodacre** [comment]
Yay travis builds!  Need to update the commit message to reflect the rimraf change.

**Comment by LiamGoodacre** [comment]
@garyb Are the `On` suffixes okay for some of the operator aliases?  (I originally used `Of` like `previewOf`, but then I came across `^..` which is flipped `toListOf` so that would have been `toListOfOf`)

**Comment by garyb** [comment]
Yep, fine by me. In other places we might have called that `toListOfFlipped`, but that's where there's no intention the operator will ever be used by name, whereas the `-On` names make sense in their own right.

**Comment by garyb** [comment]
Sorry this is languishing, I was going to hold off until we were done with the core libraries. 

Do you want me to release it as an RC now, with the caveat that it will definitely break again when we're done with core?

**Comment by LiamGoodacre** [comment]
Nah don't worry about it.  Core libraries are more important at the moment.

**Comment by garyb** [comment]
Thanks for your work on this, I cherry picked the commit and resolved the conflicts in #33.

**Pull request closed without merging**